Linting error broke CI in main

https://github.com/cmu-delphi/covidcast-indicators/runs/3070737518?check_suite_focus=true

```
. env/bin/activate; pylint delphi_utils
************* Module delphi_utils.validator.dynamic
delphi_utils/validator/dynamic.py:61:4: R0915: Too many statements (51/50) (too-many-statements)

-----------------------------------
Your code has been rated at 9.99/10

make: *** [Makefile:12: lint] Error 8
```

--- a/_delphi_utils_python/delphi_utils/validator/dynamic.py
+++ b/_delphi_utils_python/delphi_utils/validator/dynamic.py
@@ -69,14 +69,6 @@ class DynamicValidator:
         report: ValidationReport
             report to which the results of these checks will be added
         """
-        # recent_lookbehind: start from the check date and working backward in time,
-        # how many days at a time do we want to check for anomalies?
-        # Choosing 1 day checks just the daily data.
-        recent_lookbehind = timedelta(days=1)
-
-        # semirecent_lookbehind: starting from the check date and working backward
-        # in time, how many days do we use to form the reference statistics.
-        semirecent_lookbehind = timedelta(days=7)
 
         # Get 14 days prior to the earliest list date
         outlier_lookbehind = timedelta(days=14)
@@ -147,62 +139,9 @@ class DynamicValidator:
             # Check data from a group of dates against recent (previous 7 days,
             # by default) data from the API.
             for checking_date in self.params.time_window.date_seq:
-                recent_cutoff_date = checking_date - \
-                    recent_lookbehind + timedelta(days=1)
-                recent_df = geo_sig_df.query(
-                    'time_value <= @checking_date & time_value >= @recent_cutoff_date')
 
-                report.increment_total_checks()
-
-                if recent_df.empty:
-                    # If checking_date is within max_lag days, don't give error
-                    min_thres = timedelta(days = self.params.max_expected_lag.get(
-                        signal_type, self.params.max_expected_lag.get('all', 10)))
-                    if checking_date > self.params.generation_date - min_thres:
-                        report.add_raised_error(
-                            ValidationFailure("check_missing_geo_sig_date_combo",
-                                              checking_date,
-                                              geo_type,
-                                              signal_type,
-                                              "test data for a given checking date-geo type-"
-                                              "signal type combination is missing. Source data "
-                                              "may be missing for one or more dates"))
-                    continue
-
-                # Reference dataframe runs backwards from the recent_cutoff_date
-                #
-                # These variables are interpolated into the call to `api_df_or_error.query()`
-                # below but pylint doesn't recognize that.
-                # pylint: disable=unused-variable
-                reference_start_date = recent_cutoff_date - \
-                    min(semirecent_lookbehind, self.params.max_check_lookbehind) - \
-                    timedelta(days=1)
-                if signal_type in self.params.smoothed_signals:
-                    # Add an extra 7 days to the reference period.
-                    reference_start_date = reference_start_date - \
-                        timedelta(days=7)
-
-                reference_end_date = recent_cutoff_date - timedelta(days=1)
-                # pylint: enable=unused-variable
-
-                # Subset API data to relevant range of dates.
-                reference_api_df = api_df_or_error.query(
-                    "time_value >= @reference_start_date & time_value <= @reference_end_date")
-
-                report.increment_total_checks()
-
-                if reference_api_df.empty:
-                    report.add_raised_error(
-                        ValidationFailure("empty_reference_data",
-                                          checking_date,
-                                          geo_type,
-                                          signal_type,
-                                          "reference data is empty; comparative checks could not "
-                                          "be performed"))
-                    continue
-
-                reference_api_df = self.pad_reference_api_df(
-                    reference_api_df, geo_sig_df, reference_end_date)
+                recent_df, reference_api_df = self.create_dfs(
+                    geo_sig_df, checking_date, geo_type, signal_type, report)
 
                 self.check_max_date_vs_reference(
                     recent_df, reference_api_df, checking_date, geo_type, signal_type, report)
@@ -269,6 +208,69 @@ class DynamicValidator:
                                   message="date of most recent generated file seems too recent"))
 
         report.increment_total_checks()
+
+    def create_dfs(self, geo_sig_df, checking_date, geo_type, signal_type, report):
+        # recent_lookbehind: start from the check date and working backward in time,
+        # how many days at a time do we want to check for anomalies?
+        # Choosing 1 day checks just the daily data.
+        recent_lookbehind = timedelta(days=1)
+
+        # semirecent_lookbehind: starting from the check date and working backward
+        # in time, how many days do we use to form the reference statistics.
+        semirecent_lookbehind = timedelta(days=7)
+
+        recent_cutoff_date = checking_date - \
+            recent_lookbehind + timedelta(days=1)
+        recent_df = geo_sig_df.query(
+            'time_value <= @checking_date & time_value >= @recent_cutoff_date')
+
+        report.increment_total_checks()
+
+        if recent_df.empty:
+            report.add_raised_error(
+                ValidationFailure("check_missing_geo_sig_date_combo",
+                                  checking_date,
+                                  geo_type,
+                                  signal_type,
+                                  "test data for a given checking date-geo type-signal type"
+                                  " combination is missing. Source data may be missing"
+                                  " for one or more dates"))
+            continue
+
+        # Reference dataframe runs backwards from the recent_cutoff_date
+        #
+        # These variables are interpolated into the call to `api_df_or_error.query()`
+        # below but pylint doesn't recognize that.
+        # pylint: disable=unused-variable
+        reference_start_date = recent_cutoff_date - \
+            min(semirecent_lookbehind, self.params.max_check_lookbehind) - \
+            timedelta(days=1)
+        if signal_type in self.params.smoothed_signals:
+            # Add an extra 7 days to the reference period.
+            reference_start_date = reference_start_date - \
+                timedelta(days=7)
+
+        reference_end_date = recent_cutoff_date - timedelta(days=1)
+        # pylint: enable=unused-variable
+
+        # Subset API data to relevant range of dates.
+        reference_api_df = api_df_or_error.query(
+            "time_value >= @reference_start_date & time_value <= @reference_end_date")
+
+        report.increment_total_checks()
+
+        if reference_api_df.empty:
+            report.add_raised_error(
+                ValidationFailure("empty_reference_data",
+                                  checking_date,
+                                  geo_type,
+                                  signal_type,
+                                  "reference data is empty; comparative checks could not "
+                                  "be performed"))
+            continue
+
+        reference_api_df = self.pad_reference_api_df(reference_api_df, geo_sig_df, reference_end_date)
+        return (geo_sig_df, reference_api_df)
 
     def pad_reference_api_df(self, reference_api_df, geo_sig_df, reference_end_date):
         """Check if API data is missing, and supplement from test data.

--- a/_delphi_utils_python/delphi_utils/validator/dynamic.py
+++ b/_delphi_utils_python/delphi_utils/validator/dynamic.py
@@ -140,8 +140,12 @@ class DynamicValidator:
             # by default) data from the API.
             for checking_date in self.params.time_window.date_seq:
 
-                recent_df, reference_api_df = self.create_dfs(
-                    geo_sig_df, checking_date, geo_type, signal_type, report)
+                create_dfs_or_error = self.create_dfs(
+                    geo_sig_df, api_df_or_error, checking_date, geo_type, signal_type, report)
+
+                if not create_dfs_or_error:
+                    continue
+                recent_df, reference_api_df = create_dfs_or_error
 
                 self.check_max_date_vs_reference(
                     recent_df, reference_api_df, checking_date, geo_type, signal_type, report)
@@ -209,7 +213,23 @@ class DynamicValidator:
 
         report.increment_total_checks()
 
-    def create_dfs(self, geo_sig_df, checking_date, geo_type, signal_type, report):
+    def create_dfs(self, geo_sig_df, api_df_or_error, checking_date, geo_type, signal_type, report):
+        """ Create recent_df and reference_api_df from params.
+
+        Raises error if recent_df is empty.
+
+        Arguments:
+            - geo_sig_df: Pandas dataframe of test data
+            - api_df_or_error: pandas dataframe of reference data, either from the
+            COVIDcast API or semirecent data
+            - geo_type: str; geo type name (county, msa, hrr, state) as in the CSV name
+            - signal_type: str; signal name as in the CSV name
+            - report: ValidationReport; report where results are added
+
+        Returns:
+            - False if recent_df is empty, else (recent_df, reference_api_df)
+            (after reference_api_df has been padded if necessary)
+        """
         # recent_lookbehind: start from the check date and working backward in time,
         # how many days at a time do we want to check for anomalies?
         # Choosing 1 day checks just the daily data.
@@ -235,7 +255,7 @@ class DynamicValidator:
                                   "test data for a given checking date-geo type-signal type"
                                   " combination is missing. Source data may be missing"
                                   " for one or more dates"))
-            continue
+            return False
 
         # Reference dataframe runs backwards from the recent_cutoff_date
         #
@@ -267,9 +287,11 @@ class DynamicValidator:
                                   signal_type,
                                   "reference data is empty; comparative checks could not "
                                   "be performed"))
-            continue
+            return False
 
-        reference_api_df = self.pad_reference_api_df(reference_api_df, geo_sig_df, reference_end_date)
+        reference_api_df = self.pad_reference_api_df(
+            reference_api_df, geo_sig_df, reference_end_date)
+
         return (geo_sig_df, reference_api_df)
 
     def pad_reference_api_df(self, reference_api_df, geo_sig_df, reference_end_date):

--- a/_delphi_utils_python/delphi_utils/validator/dynamic.py
+++ b/_delphi_utils_python/delphi_utils/validator/dynamic.py
@@ -69,7 +69,6 @@ class DynamicValidator:
         report: ValidationReport
             report to which the results of these checks will be added
         """
-
         # Get 14 days prior to the earliest list date
         outlier_lookbehind = timedelta(days=14)
 
@@ -214,7 +213,7 @@ class DynamicValidator:
         report.increment_total_checks()
 
     def create_dfs(self, geo_sig_df, api_df_or_error, checking_date, geo_type, signal_type, report):
-        """ Create recent_df and reference_api_df from params.
+        """Create recent_df and reference_api_df from params.
 
         Raises error if recent_df is empty.
 

